engine/src/main/java/org/terasology/particles/components/ParticleSystemComponent.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.particles.components;
+
+import org.lwjgl.util.vector.Vector2f;
+import org.terasology.entitySystem.Component;
+import org.terasology.entitySystem.Owns;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.particles.events.ParticleSystemUpdateEvent;
+import org.terasology.particles.rendering.ParticleRenderer;
+import org.terasology.rendering.assets.texture.Texture;
+
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Represents a particle system. A particle system must have emitters in order to produce particles.
+ */
+public class ParticleSystemComponent implements Component {
+
+    /**
+     * Reference to the entity this component is attached to
+     */
+    public EntityRef ownerEntity;
+
+    /**
+     * The renderer that should be used to render this system.
+     */
+    private Class<ParticleRenderer> renderer;
+
+    /**
+     * The lifetime of this system
+     */
+    public float maxLifeTime = Float.POSITIVE_INFINITY;
+
+    /**
+     * Destroy this particle system when its last emitter is destroyed TODO: Unimplemented
+     */
+    public boolean destroyWhenFinished = true;
```

Comment:
I think it should be the choice of the effect's developer. 

For a small effect like the one you are mentioning disposal might not impact performance and auto-disposal wouldn't probably be a problem. In the specific case I probably still wouldn't use auto-disposal because breaking a block is a fairly common operation and the memory footprint is likely to be small anyway. For less frequent effects, say a campfire, or a comet, autodisposal might make sense.

But I'm thinking about larger effects such as rain and snow. I wouldn't want to reallocate the memory for all those particles and the associated control systems every time it has to start raining or snowing. Notice that if done GPU-side these effects might realistically have tens of thousands of particles without much loss in frame rate.